app/uk/gov/hmrc/gform/playcomponents/GformSessionCookieCryptoFilter.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.playcomponents
+
+import _root_.akka.stream.Materializer
+import cats.syntax.eq._
+import com.typesafe.config.Config
+import play.api.mvc._
+import play.api.routing.Router.RequestImplicits._
+import uk.gov.hmrc.crypto.{ Crypted, Decrypter, Encrypter, PlainText }
+import uk.gov.hmrc.gform.FormTemplateKey
+import uk.gov.hmrc.gform.controllers.CookieNames._
+import uk.gov.hmrc.gform.gformbackend.GformConnector
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ Anonymous, FormTemplate, FormTemplateId }
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+import uk.gov.hmrc.play.bootstrap.frontend.filters.crypto.{ SessionCookieCrypto, SessionCookieCryptoFilter }
+
+import scala.concurrent.{ ExecutionContext, Future }
+
+class GformSessionCookieCryptoFilter(
+  sessionCookieCrypto: SessionCookieCrypto,
+  nonanonymouscookieCryptoFilter: SessionCookieCryptoFilter,
+  anonymousCookieCryptoFilter: SessionCookieCryptoFilter,
+  val sessionBaker: SessionCookieBaker,
+  gformConnector: GformConnector,
+  config: Config
+)(implicit override val ec: ExecutionContext, override val mat: Materializer)
+    extends SessionCookieCryptoFilter {
```

Comment:
This filter shouldn't handle session cookie itself, why it extends `SessionCookieCryptoFilter`?

Reply:
At https://github.com/hmrc/gform-frontend/blob/master/app/uk/gov/hmrc/gform/playcomponents/FrontendFiltersModule.scala#L94  In bootstrap-frontend library class FrontendFilters 9th parameter as SessionCookieCryptoFilter hence extended SessionCookieCryptoFilter.

Reply:
My suggestion would be use original `SessionCookieCryptoFilter` in:
```
lazy val httpFilters: Seq[EssentialFilter] = new FrontendFilters(...) 
```
and then replace it with our `Filter`.  Something like:

```
httpFilters.map { 
  case _: SessionCookieCryptoFilter => ourFilter 
  case keep => keep
}
```